workflows/energyplus.py

```diff
@@ -203,281 +190,306 @@ def run_energyplus(self, isIP, run_directory, file_name, args, by_api):
+            # run ExpandObjects
+            try:
+                for message in self.execute_for_callback([expand_objects], output_directory):
+                    self.callback(message)
+            except subprocess.CalledProcessError:
+                self.callback("E+ FAILED")
+                return EPLaunchWorkflowResponse1(
+                    success=False,
+                    message="EnergyPlus failed for file: %s!" % full_file_path,
+                    column_data={}
+                )
+            # regardless of what happened, clean up the temp files
+            delete_if_exists(idf_path_in_output_dir)
+            delete_if_exists(idd_path_in_output_dir)
+            # if expanded.idf does not exist then we don't need to do anything, assume it just didn't have any templates
+            # but if it does exist, then this needs to be the IDF that we run, *however* we can't just leave it here
+            # as it may reference some other files back in the original directory.  I'm going to go with a new file name
+            # in the original directory of "FileName_Expanded.idf"
+            expanded_path_in_output_dir = os.path.join(output_directory, 'expanded.idf')
+            if os.path.exists(expanded_path_in_output_dir):
+                target_expanded_file_name = os.path.join(run_directory, f"{file_name_no_ext}_Expanded.idf")
+                shutil.copy(expanded_path_in_output_dir, target_expanded_file_name)
+                full_file_path = target_expanded_file_name
+
+        # Run EnergyPlus in a subdirectory
+        command_line_args += ['-d', output_directory]
+
+        # if not isIP:  # if using SI output units just use readvars CLI option
+        #     command_line_args += ['--readvars']
+
+        # add some config parameters
+        command_line_args += ['--output-prefix', file_name_no_ext, '--output-suffix', 'C']
+
+        # add in simulation control args
+        if 'weather' in args and args['weather']:
+            command_line_args += ['--weather', args['weather']]
+        else:
+            command_line_args += ['--design-day']
+
+        # and at the very end, add the file to run
+        command_line_args += [full_file_path]
+
+        # run E+
+        if by_api:
+            # if by API then find the API wrapper relative to this workflow, import it, set up a callback, and run
+            eplus_dir = Path(__file__).parent.parent.absolute()
+            sys.path.insert(0, str(eplus_dir))
+            from pyenergyplus.api import EnergyPlusAPI
+            x = lambda msg: self.callback("(E+API) " + msg.decode('utf-8', errors='ignore'))
+            # x = lambda msg: print("(E+ API) : " + str(msg))
+            api = EnergyPlusAPI()
+            state = api.state_manager.new_state()
+            api.runtime.callback_message(state, x)
+            api.runtime.set_console_output_status(state, False)
+            # cur_dir = os.getcwd()
+            # os.chdir(output_directory)
```

Comment:
Previously we would just set the current working directory and ask EnergyPlus to run.  It was no problem when only one file ran at a time.  However, the `os.getcwd` is not thread-safe, it is modifying a process attribute, so each time we come in here, this trick of convincing EnergyPlus to run in a different "current directory" falls apart, and EnergyPlus rapidly gets confused when two EnergyPlus runs happen simultaneously.

Since we cannot use that trick anymore, we have to just have EnergyPlus run in the actual current working directory, but we just tell it to output all the relevant files to the right output directory.  This is all good except for the ExpandObjects type of stuff that EnergyPlus does...which we have now had to disable and roll our own.  Not a big deal here, but an annoying thing to debug.